packages/components/charts/src/use-donut-chart.ts

```diff
@@ -0,0 +1,327 @@
+import { useTheme, type CSSUIObject, type CSSUIProps } from "@yamada-ui/core"
+import { cx, type Dict } from "@yamada-ui/utils"
+import type { ComponentPropsWithoutRef } from "react"
+import { useCallback, useMemo, useState } from "react"
+import type * as Recharts from "recharts"
+import { getClassName, getComponentProps } from "./chart-utils"
+import type {
+  CellProps,
+  ChartPropGetter,
+  DonutChartProps,
+  DonutProps,
+  RequiredChartPropGetter,
+} from "./chart.types"
+import { pieChartProperties, pieProperties } from "./rechart-properties"
+
+export type UseDonutChartOptions = {
+  /**
+   * Chart data.
+   */
+  data: CellProps[]
+  /**
+   * Props passed down to recharts `DonutChart` component.
+   */
+  chartProps?: DonutChartProps
+  /**
+   * Props for the donut.
+   */
+  donutProps?: Partial<DonutProps>
+  /**
+   * Props for the cell.
+   */
+  cellProps?: Partial<CellProps>
+  /**
+   * Determines whether each segment should have associated label.
+   *
+   * @default false
+   */
+  withLabel?: boolean
+  /**
+   * Determines whether segments labels should have lines that connect the segment with the label.
+   *
+   * @default false
+   */
+  withLabelsLine?: boolean
+  /**
+   * Controls innerRadius of the chart segments.
+   *
+   * @default '60%'
+   */
+  innerRadius?: number | string
+  /**
+   * Controls thickness of the chart segments.
+   *
+   * @default '80%'
+   */
+  outerRadius?: number | string
+  /**
+   * Controls padding between segments.
+   *
+   * @default 0
+   */
+  paddingAngle?: number
+  /**
+   * Stroke width for the chart donuts.
+   *
+   * @default 0
+   */
+  strokeWidth?: number
+  /**
+   * Controls angle at which chart starts.
+   *
+   * @default 0
+   */
+  startAngle?: number
+  /**
+   * Controls angle at which chart ends.
+   *
+   * @default 360
+   */
+  endAngle?: number
+  /**
+   * Controls fill opacity of all donuts.
+   *
+   * @default 1
+   */
+  fillOpacity?: number | [number, number]
+  /**
+   * A function to format values inside the tooltip.
+   */
+  valueFormatter?: (value: number) => string
+}
+
+type UseDonutChartProps = UseDonutChartOptions & {
+  styles: Dict<CSSUIObject>
+}
+
+export const useDonutChart = ({
+  data,
+  withLabel = false,
+  withLabelsLine = false,
```

Comment:
Change `withLabelsLine` to `withLabelLines`

Reply:
Done.